magento/magento2
# Get sitemap product images from image cache, if available

Product images in sitemap are currently always pulled directly from gallery attributes by querying the database, whilst on frontend, we're using cached versions of the (that can potentially be stored somewhere else).

This can present an issue if google doesn't want to index images in the sitemap, as their URL is different than the one on product pages.

### Description
In sitemap, product image URLs are now being pulled by using product helper, that can serve cached image, if available. This is in line to what we see on the frontend, on product page.

### Fixed Issues (if relevant)
1. magento/magento2#7504

### Manual testing scenarios
1. Create a product and set an image on it
2. Set up and generate a sitemap
3. Observe that product image URL in sitemap is direct URL to pub/media/catalog/product http://prntscr.com/eqw82x
4. Observe that images on the product page are being pulled from image cache folder http://prntscr.com/eqw90a
5. After the fix, images are being pulled via product image helper and are in line to the way product page behaves http://prntscr.com/eqw8kl

### Contribution checklist
